pkg/resource/resource.go

```diff
@@ -227,3 +236,19 @@ func IsBound(b Bindable) bool {
 func IsConditionTrue(c v1alpha1.Condition) bool {
 	return c.Status == corev1.ConditionTrue
 }
+
+// GetExternalTags returns the identifying tags to be used to tag the external
+// resource in provider API.
+func GetExternalTags(mg Managed) map[string]string {
+	tags := map[string]string{
+		ExternalResourceKindTagKey: mg.GetObjectKind().GroupVersionKind().GroupKind().String(),
```

Comment:
Can we rely on this field always being set? I recall it was being automatically cleared by some clients, apparently intentionally, which is why most of our crossplane-runtime functions that need a GVK require we pass either a GVK or a scheme to determine one. https://github.com/kubernetes-sigs/controller-runtime/pull/528 indicates that the old behaviour I described may now be gone.

Reply:
I will test it and write here. I simply relied on the fact that all those functions returned non-pointer objects.

Reply:
Verified that it does return GVK correctly.

Reply:
Nice! This might allow us to simplify some of our functions in `pkg/meta` (and possibly `pkg/resource`?) that assume they can't rely on `GetObjectKind` to actually return the kind of an object.